### DISCUSSION TITLE
在Windows下引入qt.widgetapp规则的时候，如何编译命令行程序？

### DISCUSSION THREAD

**Discussion by jingkaimori** [xmake-io/xmake#3910] [Q&A] [answered]
这一规则调用了load模块并将`gui`参数设为`true`，而在Windows下`gui=true`的程序没有命令行输入输出。

如果使用`qt.console`规则，则需要手动导入`QtWidgets`等库。

**Comment by star-hengxing** [comment, marked as answer]
```lua
add_ldflags("/subsystem:console")
```

**Comment by waruqi** [comment]
> 如果使用qt.console规则，则需要手动导入QtWidgets等库。

就是自己 add_frameworks 按需导入下，一行代码事情。。

**Reply by jingkaimori** [reply]
幸亏现在不需要同时兼容qt4和qt5，而且不需要引入系统插件，要不然还真不是一行代码的事

**Reply by waruqi** [reply]
目前 qt.widgetapp 和 qt.console 也就差这么几行的差异，https://github.com/xmake-io/xmake/blob/95df1f53c56d12b940e04b326ee05e090e3fed1a/xmake/rules/qt/xmake.lua#L89

你也可以自己参考定义自己的 qt rules 处理各种差异，然后统一使用自己的 qt rule

另外，qt.widgetapp 主要是给 gui 的用的。
